core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -216,10 +217,10 @@ class ReplicaFetcherThread(name: String,
     try {
       val clientResponse = leaderEndpoint.sendRequest(fetchRequest)
       val fetchResponse = clientResponse.responseBody.asInstanceOf[FetchResponse[Records]]
-      if (!fetchSessionHandler.handleResponse(fetchResponse)) {
+      if (!fetchSessionHandler.handleResponse(fetchResponse, fetchRequest.latestAllowedVersion())) {
```

Comment:
Should we use the latest version or fetchRequestVersion guarded by IBP?

Reply:
Good point, it should probably be along the lines of  `if (fetchRequestVersion >= 13 && !fetchData.canUseTopicIds) 12 else fetchRequestVersion` to match how it is sent below.

Reply:
I just realized that fetchRequest.latestAllowedVersion is the version we use to build the request (latest and earliest are the same.) But this is a bit confusing, so I'll probably just use the logic mentioned above.